test/e2e/service_to_service_test.go

```diff
@@ -174,22 +184,22 @@ func TestServiceToServiceCall(t *testing.T) {
 	t.Logf("helloworld internal domain is %s.", resources.Route.Status.URL.Host)
 
 	// helloworld app and its route are ready. Running the test cases now.
+	if injectA {
+		revAnno = nil
+	}
 	for _, tc := range testCases {
```

Comment:
Since we have a subtest setup anyway:

Should we ditch `testServiceToServiceCallFromZero` altogether and add an option to the test-table saying whether we want to wait for scale to 0 or not? That should cut down on code duplication in this test I think.